### PR TITLE
feat(servergroup): allow ad-hoc entity tags on deployments

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Allows a user to include an `entityTags` collection in the payload of a deploy operation, which will add those tags once the deploy completes.

**NOTE:** this is technically a breaking change - I have modified the `ServerGroupEntityTagGenerator` method signature to return a collection of tags instead of a single tag. Since this was just released a few weeks ago, and I don't know of anyone else using entity tags (much less writing their own custom generators), I'm reluctant to label it a breaking change. Will defer to the Orca owners on that.

@ajordens PTAL

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/AddServerGroupEntityTagsTask.groovy [hunk 1]
```diff
@@ -31,7 +31,7 @@ import org.springframework.stereotype.Component
 
 @Component
 @Slf4j
-class ServerGroupMetadataTagTask extends AbstractCloudProviderAwareTask implements RetryableTask {
+class AddServerGroupEntityTagsTask extends AbstractCloudProviderAwareTask implements RetryableTask {
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/AddServerGroupEntityTagsTask.groovy]: Feels like we might have grief by renaming this task ... eh @robfletcher?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/SpinnakerMetadataServerGroupTagGenerator.java [hunk 2]
```diff
@@ -61,7 +60,9 @@
     tag.put("name", "spinnaker:metadata");
     tag.put("value", value);
 
-    return tag;
+    List<Map<String, Object>> result = new ArrayList<>();
+    result.add(tag);
+    return result;
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/SpinnakerMetadataServerGroupTagGenerator.java]: `Collections.unmodifiableList(tag)` or is that messing something up downstream?

Comment (anotherchrisberry) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/SpinnakerMetadataServerGroupTagGenerator.java]: Should be fine - we do a `flatten` downstream but that ought not to matter

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupMetadataTagTask.groovy [hunk 3]
```diff
@@ -16,80 +16,13 @@
...
-  }
-
-  static class StageData {
-    @JsonProperty("deploy.server.groups")
-    Map<String, Set<String>> deployServerGroups = [:]
-  }
+@Deprecated
+class ServerGroupMetadataTagTask extends AddServerGroupEntityTagsTask {
+  // TODO: Remove after 11/11/17 - only here to smooth over initial round of deployments
+  // Just use AddServerGroupEntityTagsTask
```
Comment (anotherchrisberry) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupMetadataTagTask.groovy]: @ajordens @robfletcher doing this to address concerns about deployment and changing the class name of the task

**Pull request merged**